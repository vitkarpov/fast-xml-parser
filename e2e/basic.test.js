const fs = require('fs');
const path = require('path');
const {parse} = require('../dst/index');

describe('basic', () => {
  it('parse', () => {
    const input = fs.readFileSync(path.join(__dirname, 'inputs/basic.html'), 'utf-8');
    expect(parse(input)).toMatchSnapshot();
  });
});
