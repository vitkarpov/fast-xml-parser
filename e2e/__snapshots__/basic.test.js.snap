// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic parse 1`] = `
Node {
  "attrs": Object {},
  "children": Array [
    Node {
      "attrs": Object {},
      "children": Array [],
      "name": "!DOCTYPE html",
      "type": "DIRECTIVE",
    },
    Node {
      "attrs": Object {},
      "children": Array [
        Node {
          "attrs": Object {},
          "children": Array [
            Node {
              "attrs": Object {
                "href": "http://example.org/",
              },
              "children": Array [],
              "name": "link",
              "type": "OPENING_TAG",
            },
            Node {
              "attrs": Object {
                "charset": "utf-8",
              },
              "children": Array [],
              "name": "meta",
              "type": "OPENING_TAG",
            },
            Node {
              "attrs": Object {},
              "children": Array [
                Node {
                  "attrs": Object {},
                  "children": Array [],
                  "name": "Hello, world!",
                  "type": "TEXT",
                },
              ],
              "name": "title",
              "type": "OPENING_TAG",
            },
          ],
          "name": "head",
          "type": "OPENING_TAG",
        },
        Node {
          "attrs": Object {},
          "children": Array [
            Node {
              "attrs": Object {},
              "children": Array [
                Node {
                  "attrs": Object {},
                  "children": Array [],
                  "name": "Homepage",
                  "type": "TEXT",
                },
              ],
              "name": "h1",
              "type": "OPENING_TAG",
            },
          ],
          "name": "body",
          "type": "OPENING_TAG",
        },
      ],
      "name": "html",
      "type": "OPENING_TAG",
    },
  ],
  "name": "root",
  "type": "FRAGMENT",
}
`;
